proposals/p2483.md

```diff
@@ -0,0 +1,103 @@
+-   [Rationale](#rationale)
+-   [Alternatives considered](#alternatives-considered)
+
+<!-- tocstop -->
+
+## Abstract
+
+Use the keyword `impls` instead of `is` when writing a `where` constraint that
+at type variable needs to implement an interface or named constraint.
+
+What was previously (provisionally) written:
+
+```
+fn Sort[T: Container where .ElementType is Ordered](x: T*);
+```
+
+will now be written:
+
+```
+fn Sort[T: Container where .ElementType impls Ordered](x: T*);
+```
+
+## Problem
+
+The `is` keyword has been used as a placeholder in `where` constraints
+expressing that a type variable is required to implement an interface or named
+constraint. It has been an open question what word should be used in that
+position since its original adoption in
+[#818](https://github.com/carbon-language/carbon-lang/pull/818). Since then,
+reasons to use a different word in this position have been discovered:
+
+-   The word "is" is unspecific and could represent many different
+    relationships.
+-   This specific relationship is not symmetric.
+-   We potentially want to use `is` as a keyword for another purpose.
+-   The precedent for `is` came from Swift where `x is T` means "`x` has the
+    type `T`". With the changes to Carbon generic semantics, particularly
+    [#2360: Types are values of type `type`](https://github.com/carbon-language/carbon-lang/pull/2360),
+    that is increasingly a poor fit for what we mean by this condition.
+
+## Background
+
+The `is` keyword as a constraint operator was introduced in
+[#818: Constraints for generics (generics details 3)](https://github.com/carbon-language/carbon-lang/pull/818),
+along with the open question about how to spell it.
+
+The choice of `is` in that proposal followed
+[`is` being Swift's type check operator](https://docs.swift.org/swift-book/LanguageGuide/TypeCasting.html#ID340),
+where `x is T` is `true` if `x` has type `T`. Note that there are differences
+between the `is` operator in Swift and what we have used it for in Carbon. In
+Swift, it is used to test whether a value dynamically has a specific derived
+type, when you have a value of a base class type and are using inheritance. In
+Carbon:
+
+-   it is used on types instead of values;
+-   it is about conformance to an interface (the equivalent of Swift's
+    protocols), and not about inheritance; and
+-   it is resolved at compile time.
+
+## Proposal
+
+Use the keyword `impls` instead of `is` when writing a `where` constraint that
+at type variable needs to implement an interface or named constraint. The
+specific changes are included in
+[the same PR as this proposal](https://github.com/carbon-language/carbon-lang/pull/2483).
+
+## Rationale
+
+This proposal is working towards Carbon's
+[code that is easy to read, understand, and write](/docs/project/goals.md#code-that-is-easy-to-read-understand-and-write)
+goal:
+
+-   Examples read naturally using `impls`, commonly matching how a comment would
```

Comment:
I think this is subjective: I read "impls" as a pluralization of "impl", but don't you mean something different?

Reply:
I will rephrase to clarify what I mean.

Reply:
Sorry, but devil's advocate: is this making things easier to understand if readers understand that `impls` stands for `implements`?

What does `impl` stand for, is that also `implements` but with a different meaning?

Reply:
`impl` comes from Rust, where I think it is short for "implementation," as in "implementation of `Display` for `i32`." This matches the Carbon convention where the introducer keywords correspond to nouns ("implementation", "function", "variable", "class") not verbs ("implement" or "implements").